tests/integration/targets/docker_config/tasks/test_docker_config.yml

```diff
@@ -226,6 +227,99 @@
       that:
         - output.failed
 
+# template_driver tests
+
+  - when: docker_py_version is version('5.0.3', '>=') and docker_api_version is version('1.37', '>=')
+    block:
+
+    - name: Create regular config
+      docker_config:
+        name: db_password
+        data: opensesame!
+        state: present
+
+    - name: Update config with template_driver
+      docker_config:
+        name: db_password
+        data: opensesame!
+        template_driver: golang
+        state: present
+      register: output
+
+    - name: Assert config was updated
+      assert:
+        that:
+           - output is changed
+
+    - name: Invalid template_driver
+      docker_config:
+        name: db_password
+        data: opensesame!
+        template_driver: "not a template driver"
+        state: present
+      ignore_errors: yes
+      register: output
+
+    - name: Assert failure when called with invalid template_driver
+      assert:
+        that:
+           - 'output.failed'
+           - 'output.msg == "value of template_driver must be one of: golang, got: not a template driver"'
+
+    - name: Create config again
+      docker_config:
+        name: db_password
+        data: opensesame!
+        template_driver: golang
+        state: present
+      register: output
+
+    - name: Assert create config is idempotent
+      assert:
+        that:
+           - output is not changed
+
+    # data is the docker swarm's name
+    - name: Update config with template data
+      docker_config:
+        name: db_password
+        data: "{{ '{{' }} .Service.Name {{ '}}' }}"
+        template_driver: golang
+        state: present
+      register: output
+
+    - name: Inspect config
+      command: "docker config inspect {{ output.config_id }}"
+      register: inspect
+
+    - name: Show inspection result
+      debug:
+        var: inspect
+
+    - name: assert config creation succeeded
+      assert:
+        that:
+           - "'db_password' in inspect.stdout"
+           - "'ansible_key' in inspect.stdout"
+           - "'\"Data\": \"default\"' in inspect.stdout"
```

Comment:
BTW, this never passes, since (also confirmed by the API docs) `Data` is always the url-safe-Base64 encoded original data. So here you get `e3sgLlNlcnZpY2UuTmFtZSB9fQ==` and not `default`.

Reply:
Right ok nice work